src/modules/core/components/price-field/price-field.tsx

```diff
@@ -0,0 +1,90 @@
+import { type FC, type ChangeEvent, Fragment } from 'react';
+import { useBoolean } from 'usehooks-ts';
+import clsx from 'clsx';
+// components
+import {
+  Divider,
+  Icon,
+  Popover,
+  TextField,
+  Typography,
+} from '@/modules/core/components';
+// constants
+import {
+  currencies,
+  currencyMeta,
+} from '@/modules/core/constants/currency.constants';
+// types
+import type { Currency } from '@/modules/core/types/currency.types';
+
+import type { PriceFieldProps } from './price-field.interface';
+import styles from './price-field.module.scss';
+
+export const PriceField: FC<PriceFieldProps> = ({
```

Comment:
Can we select all when focusing on this field, as we have leading zero when using this field on the quizz, also it would be nice to format numbers to show thousands, like this 23,000 instead of 23000.

Reply:
done